### PR TITLE
Resolve image names with default transport in from command

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: pull.go [hunk 1]
```diff
@@ -146,11 +146,11 @@ func pullImage(ctx context.Context, store storage.Store, imageName string, optio
 	srcRef, err := alltransports.ParseImageName(spec)
 	if err != nil {
 		if options.Transport == "" {
-			options.Transport = DefaultTransport
+			options.Transport = util.DefaultTransport
 		}
 		logrus.Debugf("error parsing image name %q, trying with transport %q: %v", spec, options.Transport, err)
 		transport := options.Transport
-		if transport != DefaultTransport {
+		if transport != util.DefaultTransport {
```
Comment (TomSweeneyRedHat) [inline pull.go]: Can you add the error message from 1119 that's new to pull.go in this?  Or your variant of the message, then we can close off @bparees issue too.

File: util/util.go [hunk 2]
```diff
@@ -89,6 +93,7 @@ func ResolveName(name string, firstRegistry string, sc *types.SystemContext, sto
 		}
 	}
 
+	name = strings.TrimPrefix(name, DefaultTransport)
```
Comment (TomSweeneyRedHat) [inline util/util.go]: In my review I'd done this at one point, but was concerned that the DefaultTransport might not be the only valid transport at some point and to trust the user.  Was I over thinking?  This did work when I tested it though.

Comment (rhatdan) [reply util/util.go]: Well the Default Transport is the only one that looks like `NAME://`  All other transports are `NAME:` and are handled separately.

So I don't see this as an issue.

File: pull.go (hunk 1 above)
Comment (rhatdan) [reply pull.go]: Done

**Comment by rhatdan** [comment]
Addresses: #1117

**Comment by rhatdan** [comment]
Fixes https://github.com/containers/buildah/issues/1057

File: util/util.go (hunk 2 above)
Comment (TomSweeneyRedHat) [reply util/util.go]: Right, but this code only works for "docker://", would it be valid to have something else there at some point like "oci://"?  Also if they put in "Docker://" they'll get a bad reference message which is probably OK, not sure if you want to lowercase name or not.

File: pull.go [hunk 3]
```diff
@@ -201,6 +201,7 @@ func pullImage(ctx context.Context, store storage.Store, imageName string, optio
 
 	logrus.Debugf("copying %q to %q", spec, destName)
 	if _, err := cp.Image(ctx, policyContext, destRef, srcRef, getCopyOptions(options.ReportWriter, srcRef, sc, destRef, nil, "")); err != nil {
+		logrus.Debugf("error copying src image [%q] to dest image [%q] err: %v", spec, destiName, err)
```
Comment (TomSweeneyRedHat) [inline pull.go]: Crud, my bad, builds are failing due to this.  Change "destiName" to "destName", no "i".

**Comment by rhatdan** [comment]
@rh-atomic-bot r=TomSweeneyRedHat

**Comment by rhatdan** [comment]
Since github is not working properly, I will manually merge.

**Pull request merged**